pkg/resource/deploy/plan_executor.go

```diff
@@ -372,14 +426,16 @@ func (pe *planExecutor) refresh(callerCtx context.Context, opts Options, preview
 		}
 
 		if new == nil {
-			contract.Assert(old.Custom)
-			contract.Assert(!providers.IsProviderType(old.Type))
+			if refresh {
+				contract.Assert(old.Custom)
+				contract.Assert(!providers.IsProviderType(old.Type))
+			}
 			continue
 		}
 
 		// Remove any deleted resources from this resource's dependency list.
 		if len(new.Dependencies) != 0 {
-			deps := make([]resource.URN, 0, len(new.Dependencies))
+			deps := []resource.URN{}
```

Comment:
Why change this?

Reply:
fell out from some refactorings.  but ended up going back, so i reverted this.